typescript/sdk/src/providers/providerBuilders.ts

```diff
@@ -109,13 +111,30 @@ export function defaultCosmJsWasmProviderBuilder(
   };
 }
 
+export function defaultZKSyncProviderBuilder(
+  rpcUrls: RpcUrl[],
+  network: providers.Networkish,
+): ZKSyncProvider {
+  assert(rpcUrls.length, 'No RPC URLs provided');
+  const url = rpcUrls[0].http;
+  const provider = new ZKProvider(url, network);
+  return { type: ProviderType.ZKSync, provider };
+}
+
 // Kept for backwards compatibility
 export function defaultProviderBuilder(
   rpcUrls: RpcUrl[],
   _network: number | string,
 ): providers.Provider {
   return defaultEthersV5ProviderBuilder(rpcUrls, _network).provider;
 }
+// Kept for backwards compatibility
+export function defaultZKProviderBuilder(
```

Comment:
Why would we need this for backwards compat when it doesn't currently exist? You can remove this

Reply:
we use it in MultiProvider, will remove this comment